api/base/serializers.py

```diff
@@ -222,13 +222,25 @@ def __init__(self, related_view=None, related_view_kwargs=None, self_view=None,
         self.view_kwargs = {'related': related_kwargs, 'self': self_kwargs}
         self.related_meta = related_meta
         self.self_meta = self_meta
+        self.always_embed = always_embed
         assert (related_view is not None or self_view is not None), 'Self or related view must be specified.'
         if related_view:
             assert related_kwargs is not None, 'Must provide related view kwargs.'
+            assert isinstance(related_kwargs, dict), "Related view kwargs must have format {'lookup_url_kwarg: lookup_field}."
         if self_view:
             assert self_kwargs is not None, 'Must provide self view kwargs.'
+            assert isinstance(self_kwargs, dict), "Self view kwargs must have format {'lookup_url_kwarg: lookup_field}."
+
+        view_name = related_view
+        if view_name:
+            lookup_kwarg = related_kwargs.keys()[0]
```

Comment:
That is brittle because it relies so much on the implementation details of the fields. Those lines will only work if and only if every field that handles multiple kwargs has a `kwargs` attribute. This is difficult to enforce. `RelationshipField`, for one, does not have a `kwargs` attribute.

Moving the responsibility of resolving a view to the Field classes solves two problems:

- Accounts for the fact that different fields may have different ways of handling view args/kwargs (e.g. some have 1 kwarg, others have multiple, some use a `lookup_kwarg` attribute, others use `'var_name': <var_value>`).
- Requiring fields to implement a `resolve` method is easier to enforce than trying to handle all the above scenarios in JSONAPIBaseView#_get_embed_partial. Calling `field.resolve(..)` will raise an AttributeError if it isn't implemented, which is a good thing. The lines you linked to fail silently for fields that aren't supported (e.g. RelationshipField).